openedx_learning/apps/authoring/publishing/contextmanagers.py

```diff
@@ -0,0 +1,157 @@
+"""
+Context Managers for internal use in the publishing app.
+
+Do not use this directly outside the publishing app. Use the public API's
+bulk_draft_changes_for instead (which will invoke this internally).
+"""
+from __future__ import annotations
+
+from contextvars import ContextVar
+from datetime import datetime, timezone
+from typing import Callable
+
+from django.db.transaction import Atomic
+
+from .models import DraftChangeLog
+
+
+class DraftChangeLogContext(Atomic):
+    """
+    Context manager for batching draft changes into DraftChangeChangeLogs.
+
+    🛑 This is a PRIVATE implementation. Outside of the publishing app, clients
+    should use the bulk_draft_changes_for() API call instead of using this
+    manager directly, since this is a bit experimental and the implementation
+    may shift around a bit.
+
+    The main idea is that we want to create a context manager that will keep
```

Comment:
@ormsbee @bradenmacdonald I think that much of this docstring--everything except the internal implementation details--should be moved to the public `bulk_draft_changes_for` docstring. Let me know if you disagree, otherwise I'll make that change.

Reply:
@kdmccormick I don't know if I would move quite as much as you're suggesting, but certainly the `bulk_draft_changes_for` docstring is incomplete at the moment and needs more of this information + examples.